data-collection/src/test/java/integration/realprojects/ApacheCommonsCliIntegrationTest.java

```diff
@@ -27,12 +29,52 @@ protected String track() {
 		return "src/java/org/apache/commons/cli/Option.java";
 	}
 
+	/*
+	Test the isSubclass boolean
+	 */
+	@Test
+	public void isSubclass() {
+		List<Yes> yesList = session.createQuery("From Yes where (className = :className1 or className = :className2) and project = :project")
+				.setParameter("className1", "org.apache.commons.cli.HelpFormatter")
+				.setParameter("className2", "org.apache.commons.cli.HelpFormatter.StringBufferComparator")
+				.setParameter("project", project)
+				.list();
+
+		Assert.assertEquals(10, yesList.size());
+		for (Yes yes : yesList){
+			switch(yes.getClassName()){
+				case "org.apache.commons.cli.HelpFormatter.StringBufferComparator":
+					Assert.assertTrue(yes.getClassMetrics().isSubclass());
+					break;
+				default:
+					Assert.assertFalse(yes.getClassMetrics().isSubclass());
+			}
+		}
+
+		List<No> noList = session.createQuery("From No where (className = :className1 or className = :className2) and project = :project")
+				.setParameter("className1", "org.apache.commons.cli.HelpFormatter")
+				.setParameter("className2", "org.apache.commons.cli.HelpFormatter.StringBufferComparator")
+				.setParameter("project", project)
+				.list();
+
+		//TODO: figure out why the where condition doesn't work
```

Comment:
Not working? Bizarre!

Printing the SQL that Hibernate is generating might give you a hint. A SO post that might help (although we use XML instead of properties for the log4j): https://stackoverflow.com/questions/1710476/how-to-print-a-query-string-with-parameter-values-when-using-hibernate

Reply:
I could resolve this issue, I hand an error in my SQl statement.